staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -212,27 +210,6 @@ func newRequestForProxy(location *url.URL, req *http.Request) (*http.Request, co
 	return newReq, cancelFn
 }
 
-// maybeWrapForConnectionUpgrades wraps the roundtripper for upgrades.  The bool indicates if it was wrapped
-func maybeWrapForConnectionUpgrades(restConfig *restclient.Config, rt http.RoundTripper, req *http.Request) (http.RoundTripper, bool, error) {
-	if !httpstream.IsUpgradeRequest(req) {
-		return rt, false, nil
-	}
-
-	tlsConfig, err := restclient.TLSConfigFor(restConfig)
-	if err != nil {
-		return nil, true, err
-	}
-	followRedirects := utilfeature.DefaultFeatureGate.Enabled(genericfeatures.StreamingProxyRedirects)
-	requireSameHostRedirects := utilfeature.DefaultFeatureGate.Enabled(genericfeatures.ValidateProxyRedirects)
-	upgradeRoundTripper := spdy.NewRoundTripper(tlsConfig, followRedirects, requireSameHostRedirects)
-	wrappedRT, err := restclient.HTTPWrappersForConfig(restConfig, upgradeRoundTripper)
```

Comment:
doesn't this call set up the front proxy headers for the connections we re-establish?

Reply:
Is the `front proxy header` the same as the auth proxy headers that are set [here](https://github.com/kubernetes/kubernetes/blob/224bcf63f252658fffc418a1d90530c6867f8b3c/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go#L166-L174)? Since we now use the same `proxyRoundTripper` to handle both upgrade/non-upgrade requests, those auth proxy headers are applied to upgrade requests as well.

Reply:
Thanks.  yes, that's the one I was looking for.